src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -193,197 +331,93 @@ private ExecutorInfo createExecutor(String taskIdName, String nodeName, String e
-      for (Offer offer : offers) {
-        driver.declineOffer(offer.getId());
-      }
-      return;
-    }
-
-    if (liveState.getNameNodes().size() == 0 && liveState.getJournalNodes().size() == 0) {
-      log.info("No NameNodes or JournalNodes found.  Collecting offers until we have sufficient "
-          + "capacity to launch.");
-      for (Offer offer: offers) {
-          pendingOffers.put(offer.getId(), offer);
-      }
-        
-      if (!initializingCluster && pendingOffers.size() >= (HDFSConstants.TOTAL_NAME_NODES + conf.getJournalNodeCount())) {
-        log.info(String.format("Launching initial nodes with %d pending offers",
-            pendingOffers.size()));
-        initializingCluster = true;
-        launchInitialJournalNodes(driver, pendingOffers.values());
-        launchInitialNameNodes(driver, pendingOffers.values());
-        // Decline any remaining offer
-        for (OfferID offerID : pendingOffers.keySet()) {
-          driver.declineOffer(offerID);
-        }
-        pendingOffers.clear();
-      }
-      return;
-    }
-
-    List<Offer> remainingOffers = new ArrayList<>();
-    remainingOffers.addAll(offers);
-
-    if (initializingCluster) {
-      log.info(String.format("Declining remaining %d offers pending initialization",
-          remainingOffers.size()));
-      for (Offer offer : remainingOffers) {
-        driver.declineOffer(offer.getId());
-      }
-      return;
-    }
-
-    offers = remainingOffers;
-    remainingOffers = new ArrayList<>();
-
-    // Check to see if we can launch some DataNodes
-    for (Offer offer : offers) {
-      if (liveState.notInDfsHosts(offer.getSlaveId().getValue())) {
-        launchDataNode(driver, offer);
-      } else {
-        remainingOffers.add(offer);
-      }
-    }
-
-    // Decline remaining offers
-    log.info(String.format("Declining %d offers", remainingOffers.size()));
-    for (Offer offer : remainingOffers) {
-      driver.declineOffer(offer.getId());
+  private boolean maybeLaunchDataNode(SchedulerDriver driver, Offer offer) {
+    if (persistentState.dataNodeRunningOnSlave(offer.getHostname()) ||
+        persistentState.nameNodeRunningOnSlave(offer.getHostname()) ||
+        persistentState.journalNodeRunningOnSlave(offer.getHostname())) {
+      log.info(String.format("Already running hdfs task on %s", offer.getHostname()));
+      return false;
+    } else {
+      launchNode(
+          driver,
+          offer,
+          HDFSConstants.DATA_NODE_ID,
+          Arrays.asList(HDFSConstants.DATA_NODE_ID),
+          HDFSConstants.NODE_EXECUTOR_ID);
+      return true;
     }
   }
 
-  @Override
-  public void slaveLost(SchedulerDriver driver, SlaveID slaveId) {
-    log.info("Slave lost slaveId=" + slaveId.getValue());
+  private void sendMessageTo(SchedulerDriver driver, TaskID taskId, SlaveID slaveID, String message) {
+    log.info(String.format("Sending message '%s' to taskId=%s, slaveId=%s", message,
```

Comment:
Might want to lower the verbosity on some of these log messages in the future, only log the noisy lines in log.debug.